src/rt/sched/behaviour.h

```diff
@@ -20,6 +22,13 @@ namespace verona::rt
       Be* body = behaviour->get_body<Be>();
       (*body)();
 
+      if (behaviour_yielded)
+      {
+        behaviour_yielded = false;
+        work->yielded = true;
```

Comment:
Schedule work here, then you don't need to add the field to `work`.
```suggestion
        Scheduler::schedule(work);
```

Reply:
The scheduler is designed that the work item is responsible for its own deallocation, so that the work item can reschedule itself, e.g.:

https://github.com/microsoft/verona-rt/blob/ab42a9f67b73273664ef31c58bd2dc0c10cddd5a/src/rt/sched/core.h#L36-L44